packages/react/src/xcss-prop/index.ts

```diff
@@ -145,23 +145,31 @@ export type XCSSAllPseudos = CSSPseudos;
  * <Component xcss={styles.text} />
  * ```
  *
- * To concatenate and conditonally apply styles use the {@link cssMap} {@link cx} functions.
+ * To concatenate and conditionally apply styles use the {@link cssMap} {@link cx} functions.
  */
 export type XCSSProp<
-  TAllowedProperties extends keyof StrictCSSProperties,
+  TAllowedProperties extends keyof StrictCSSProperties | StrictCSSProperties, // extend StrictCSSProperties with CSS vars.
```

Comment:
This won't work deeply though, you cannot restrict pseudos or anything, I don't think they will ever be allowed.

This is how I did it before, to swap out the entire `TSchema` thing: https://github.com/atlassian-labs/compiled/pull/1637/files#diff-1d4b97255cb11e9b25307047d51c7f2c1d94052d4626cc8e19c410a8b0444788R431 and unfortunately that _might_ be the only way to make that work without a major refactor.

I do like the idea of just taking a schema here upfront, _maybe_ there's a world in which the type becomes
```tsx
import type { StrictXCSSProp, DesignTokenSchema } from '@atlaskit/css';

type IconSchema = Omit<DesignTokenSchema, …> & { color: … | 'currentColor' };

function Icon (props: {
  xcss?: StrictXCSSProp<IconSchema, { requiredProperties: 'color' }>
}) { 
```

Reply:
…but that still doesn't work in the test I referenced, so not sure, didn't dive in terribly deep.

Reply:
@kylorhall-atlassian good catch on the filtering issue; had another look over, added the filtering logic for pseudos, auto-inferred the value for the second generic, and added more tests around pseudos.

Re: working deeply, what behaviour are you looking for? Pseudos seem to be working:

 ```tsx
 interface MyComponentProps {
   xcss: XCSSProp<{
     // Properties can be optional or required, and can have strict types
     color: 'color.text.red' | 'color.text.blue';
     borderColor?: 'color.border';
     // Pseudos are always optional, but you can define what pseudos are valid
     '&:hover': {
       // Setting a value in a pseudo lets you further refine the valid values
       color: 'color.text.green';
     };
     '&:focus': {
       // setting color to `never` prevents it from being set
       color: never;
     };
   }>
 }
 ```

With some limitations:
- You can't mark a pseudo as required (this isn't possible with XCSSProp as I understand, it just seems intuitive to be able to do this when you're passing in a schema as the first generic)
- If a pseudo _is_ optional, you can't customise the properties it accepts (not sure why)  
- If you want to prevent pseudos from accepting certain values, you need to explicitly re-type them, or set them to `never`

>You will have to actually change the underlying implementation, just the loose one probably isn't sufficient

Could you clarify this a bit? Currently this PR is just intended for the loose variant of XCSSProp - are you requesting that the functionality be extended to the strict version for consistency?

Reply:
Got it, maybe this `Filter<…>` fixed my initial issue with deep pseudos blowing up—now it _mostly works_.

Originally, I figured we'd have to touch `Internal$XCSSProp` (and I still think so if we want to touch on any edge-cases)—that's the main reason for the comment.  A few minor deviations isn't ideal as the APIs are already incongruent, but if we go so deep (which I guess I expected), it feels painful.

> If you want to prevent pseudos from accepting certain values
Testing your above doesn't work though—it expects `'&:focus'` to have `{ readonly color: never; }` because `color: 'red'` becomes a `requiredProperties` and passed down into `XCSSPseudo<…>`, so that doesn't quite work.
```tsx
<CSSPropComponent
  xcss={{
    color: 'color.text.red',
    '&:hover': { color: 'color.text.green', borderColor: 'color.border' },
    '&:focus': { borderColor: 'color.border' }, // errors
  }}
/>
```
I think that actually just works fine in media queries, because we don't deeply require them, IIRC…